table/server/master/src/main/java/alluxio/master/table/DbConfig.java

```diff
@@ -11,25 +11,205 @@
+  @JsonDeserialize(using = BypassTableEntryDeserializer.class)
+  public static class BypassTableEntry {
+    private final String mTableName;
+    private final Set<String> mPartitions;
+
+    /**
+     * @param tableName table name
+     * @param partitions partition names
+     */
+    @JsonCreator
+    public BypassTableEntry(@JsonProperty("table") String tableName,
+                            @JsonProperty("partitions") Set<String> partitions) {
+      Preconditions.checkArgument(!tableName.isEmpty(), "empty table name");
+      mTableName = tableName;
+      mPartitions = partitions;
+    }
+
+    /**
+     * @return table name
+     */
+    public String getTable() {
+      return mTableName;
+    }
+
+    /**
+     * @return partition names
+     */
+    public Set<String> getPartitions() {
+      return mPartitions;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+      if (this == other) {
+        return true;
+      } else if (other == null) {
+        return false;
+      } else if (getClass() != other.getClass()) {
+        return false;
+      }
+      BypassTableEntry entry = (BypassTableEntry) other;
+      return Objects.equals(mTableName, entry.mTableName);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hashCode(mTableName);
+    }
   }
 
   /**
-   * @return the bypass set
+   * Deserializer of BypassTableEntry
+   *
+   * Enables flexible syntax: either a single table name can be specified, and all belonging
+   * partitions will be bypassed;
+   * or an object of form
+   * {"table": "tableName", "partitions": ["part1", "part2"]}
+   * can be used, and individual partitions can be specified.
    */
-  public Set<String> getBypassSet() {
-    return mBypassSet;
+  public static class BypassTableEntryDeserializer extends JsonDeserializer<BypassTableEntry> {
+    @Override
+    public BypassTableEntry deserialize(JsonParser jp, DeserializationContext cxt)
+        throws IOException, JsonProcessingException {
+      ObjectMapper mapper = (ObjectMapper) jp.getCodec();
+      JsonNode node = mapper.readTree(jp);
+      String tableName;
+      Set<String> partitions;
+      if (node.isTextual()) {
+        // single table name, all partitions are bypassed
+        tableName = node.asText();
+        partitions = new HashSet<>();
+      } else if (node.isObject()) {
+        // a {"table": "table", "partitions": ["part1", "part2"]} object
+        if (!node.hasNonNull("table")) {
+          throw new JsonParseException(mapper.treeAsTokens(node), "missing table name");
+        }
+        tableName = node.get("table").asText();
+        JsonNode partitionsList = node.get("partitions");
+        partitions = mapper.convertValue(partitionsList,  new TypeReference<Set<String>>() {});
+        if (partitions == null) {
+          partitions = new HashSet<>();
+        }
+      } else {
+        throw new JsonParseException(mapper.treeAsTokens(node), "invalid syntax");
```

Comment:
nit: for code readability, try to do error-handling first before processing different cases.